lms/djangoapps/course_blocks/transformers/user_partitions.py

```diff
@@ -97,36 +97,35 @@ def transform_block_filters(self, usage_info, block_structure):
             transformer_block_field = block_structure.get_transformer_block_field(
                 block_key, self, 'merged_group_access'
             )
-            access_denying_partition_id = transformer_block_field.get_access_denying_partition(
+            access_denying_partition_ids = transformer_block_field.get_access_denying_partitions(
                 user_groups
             )
-            access_denying_partition = get_partition_from_id(user_partitions, access_denying_partition_id)
-
-            if access_denying_partition:
-                user_group = user_groups.get(access_denying_partition.id)
-                allowed_groups = transformer_block_field.get_allowed_groups()[access_denying_partition.id]
-                access_denied_message = access_denying_partition.access_denied_message(
-                    block_key, user, user_group, allowed_groups
-                )
+            access_denying_reasons = []
+            access_denying_messages = []
+            for partition_id in access_denying_partition_ids:
+                access_denying_partition = get_partition_from_id(user_partitions, partition_id)
+
+                if access_denying_partition:
+                    user_group = user_groups.get(access_denying_partition.id)
+                    allowed_groups = transformer_block_field.get_allowed_groups()[access_denying_partition.id]
+                    access_denied_message = access_denying_partition.access_denied_message(
+                        block_key, user, user_group, allowed_groups
+                    )
+                    access_denying_reasons.append(access_denying_partition.name)
+                    access_denying_messages.append(access_denied_message)
+
+            if access_denying_reasons and not access_denying_messages:
```

Comment:
What's this `if` about? Don't both arrays always get added to at same time? Same with separate `ifs` below for each array.

Reply:
Oh, yeah, I should only be adding the message to the list of messages if it exists. The `.name`/`access_denying_reason` always exists, but the message doesn't.